next-app/lib/storageFuncs.js

```diff
@@ -52,63 +76,60 @@ function deleteFromStorage (wordToDelete, place, template, time) {
         };
         
         if (existingData.length == 0) { 
-            ls.removeItem(place);
+            chrome.storage.local.remove(place);
             // templateBuilder(template); 
         } else { 
-            ls.setItem(place, JSON.stringify(existingData)); 
+            chrome.storage.local.set({[place]: existingData}); 
             // templateBuilder(template, existingData.reverse());
         }
     }
 };
 
-function savedOrganizer (searchWord) {
+async function savedOrganizer (searchWord) {
     console.log("savedOrganizer");
-    if (typeof window !== 'undefined' && window.localStorage) {
-        const ls = localStorage;
-
-        var historyItems;
-        var favouriteItems;
-        if (JSON.parse(ls.getItem("history"))) { historyItems = JSON.parse(ls.getItem("history")); } else { historyItems = []; };
-        if (JSON.parse(ls.getItem("favourites"))) { favouriteItems = JSON.parse(ls.getItem("favourites")); } else { favouriteItems = []; };
+    if (typeof window !== 'undefined') {
+        const storedHistoryItems = await chrome.storage.local.get("history");
+        const storedFavouriteItems = await chrome.storage.local.get("favourites");
+        var historyItems = Object.values(storedHistoryItems).length !== 0 ? storedHistoryItems["history"] : [];
+        var favouriteItems =  Object.values(storedFavouriteItems).length !== 0 ? storedFavouriteItems["favourites"] : [];
```

Comment:
why do you use var here?

Reply:
i think it was in your function from before because i do not use var in any case - i will switch to const/let here

Reply:
Thanks a lot, my bad